src/test_helper_functions.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * test_output.cpp
+ *
+ * Copyright (C) 2023 charlie-foxtrot
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "test_base_class.h"
+
+#include "helper_functions.h"
+
+using namespace std;
+
+class HelperFunctionsTest : public TestBaseClass {
+protected:
+
+	void SetUp(void) {
+		TestBaseClass::SetUp();
+	}
+
+	void create_file(const string &filepath) {
+		fclose(fopen(filepath.c_str(), "wb"));
+		EXPECT_TRUE(file_exists(filepath));
+	}
+
+};
+
+TEST_F(HelperFunctionsTest, rename_file_if_exists_exists)
+{
+	string starting_path = temp_dir + "/starting_path";
+	string ending_path = temp_dir + "/ending_path";
+	create_file(starting_path);
+	EXPECT_EQ(rename_file_if_exists(starting_path, ending_path), 0);
+	EXPECT_FALSE(file_exists(starting_path));
+	EXPECT_TRUE(file_exists(ending_path));
+}
+
+TEST_F(HelperFunctionsTest, rename_file_if_exists_doesnt_exist)
+{
+	string starting_path = temp_dir + "/starting_path";
+	string ending_path = temp_dir + "/ending_path";
+	EXPECT_EQ(rename_file_if_exists(starting_path, ending_path), 0);
```

Comment:
shouldn't this return non-zero?

Reply:
its the "if exists part", so the function returns success if the file is renamed or if rename filed because the input file didn't exist.  will add a test case for failure

Reply:
started going down a rabbit hole here, looks like return value isn't behaving as expected elsewhere in the code, going to roll-back this change and address under a different PR